resources/fetchers/iam_factory.go

```diff
@@ -0,0 +1,63 @@
+package fetchers
+
+import (
+	"github.com/elastic/cloudbeat/resources/providers/awslib"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/cloudbeat/resources/manager"
+)
+
+const (
+	IAMType = "aws-iam"
+)
+
+func init() {
+
+	manager.Factories.ListFetcherFactory(IAMType, &IAMFactory{
+		extraElements: getIamExtraElements,
+	})
+}
+
+type IAMFactory struct {
+	extraElements func() (IAMExtraElements, error)
+}
+
+type IAMExtraElements struct {
+	iamProvider awslib.IAMRolePermissionGetter
+}
+
+func (f *IAMFactory) Create(c *common.Config) (fetching.Fetcher, error) {
+	cfg := IAMFetcherConfig{}
```

Comment:
Add info log on factory creations (all the factories that are part of this the pr)